src/test/java/org/opentripplanner/pricing/transit/ticket/pattern/FareSwitchPatternTest.java

```diff
@@ -0,0 +1,181 @@
+        Stop futureStop = new Stop();
+
+        FareSwitch validatedSwitch = new FareSwitch(previousRoute, futureRoute, previousStop, futureStop);
+
+        assertFalse(pattern.matches(validatedSwitch));
+    }
+
+    @Test
+    public void shouldMatchWithNullConstraints() {
+        FareSwitchPattern pattern = new FareSwitchPattern(null, null, null, null, false);
+
+        Route previousRoute = new Route();
+        previousRoute.setId(new FeedScopedId());
+        previousRoute.setShortName("81");
+        Stop previousStop = new Stop();
+        Route futureRoute = new Route();
+        futureRoute.setId(new FeedScopedId());
+        futureRoute.setShortName("512");
+        Stop futureStop = new Stop();
+
+        FareSwitch validatedSwitch = new FareSwitch(previousRoute, futureRoute, previousStop, futureStop);
+
+        assertTrue(pattern.matches(validatedSwitch));
+    }
+
+    @Test
+    public void shouldMatchFareSwitchingWithStopConstraints() {
+        futureStopPattern.addConstraint(StopPattern.StopAttribute.NAME, Pattern.TextOperator.IN, "Śmigłowca");
+
+        FareSwitchPattern pattern = new FareSwitchPattern(previousRoutePattern, futureRoutePattern, previousStopPattern, futureStopPattern, false);
+
+        Route previousRoute = new Route();
+        previousRoute.setId(new FeedScopedId());
+        previousRoute.setShortName("81");
+        Stop previousStop = new Stop();
+        previousStop.setId(new FeedScopedId());
+        Route futureRoute = new Route();
+        futureRoute.setId(new FeedScopedId());
+        futureRoute.setShortName("3");
+        Stop futureStop = new Stop();
+        futureStop.setId(new FeedScopedId());
+        futureStop.setName("Śmigłowca");
+
+        FareSwitch validatedSwitch = new FareSwitch(previousRoute, futureRoute, previousStop, futureStop);
+
+        assertTrue(pattern.matches(validatedSwitch));
+    }
+
+    @Test
+    public void shouldNotMatchFareSwitchingWithStopConstraints() {
+        futureStopPattern.addConstraint(StopPattern.StopAttribute.NAME, Pattern.TextOperator.IN, "Śmigłowca");
+
+        FareSwitchPattern pattern = new FareSwitchPattern(previousRoutePattern, futureRoutePattern, previousStopPattern, futureStopPattern, false);
+
+        Route previousRoute = new Route();
+        previousRoute.setId(new FeedScopedId());
+        previousRoute.setShortName("81");
+        Stop previousStop = new Stop();
+        previousStop.setId(new FeedScopedId());
+        Route futureRoute = new Route();
+        futureRoute.setId(new FeedScopedId());
+        futureRoute.setShortName("3");
+        Stop futureStop = new Stop();
+        futureStop.setId(new FeedScopedId());
+        futureStop.setName("Dworzec Gdański");
+
+        FareSwitch validatedSwitch = new FareSwitch(previousRoute, futureRoute, previousStop, futureStop);
+
+        assertFalse(pattern.matches(validatedSwitch));
+    }
+
+    //TODO: check rules with stop rules as well!!!
```

Comment:
Is it a comment to resolve in future PRs regarding the same issue? If yes, then please add `// TODO VMP-149: ...`

Reply:
This one is indeed already resolved - I have removed the comment